app/views/cart/show.html.erb

```diff
@@ -46,7 +46,13 @@
 </table>
   <p>Total: <%= number_to_currency(cart.total) %></p>
   <p><%= link_to "Empty Cart", "/cart", method: :delete %></p>
-  <p><%= link_to "Checkout", "/orders/new", method: :get %></p>
+  <section id="bottom-text">
+    <% if current_user %>
+      <p><%= link_to "Checkout", "/orders/new", method: :get %></p>
+    <% else %>
+      <p>You must <%= link_to "register", "/register" %> or <%= link_to "log in", "/login" %> to finish the checkout process</p>
```

Comment:
I like how it gives you the links within the message to either register or login